### PR TITLE
Add generic guard for featuretoggles

### PR DESCRIPTION
**Pull request by ronaldmansveld**
Resolves # . (This is applicable only if this pull request relates to a GitHub issue, delete the line otherwise)

Notes:
*
*
*

### CODE REVIEW

**Review by paul-graham — APPROVED** [review]
Looks good to me. Even though my version number was stolen again. :)

**Review by udaydenduluri33 — COMMENTED** [review]
LGTMr
Thanks for the implementation. 
Can you add few unit tests please? Thanks

**Review by udaydenduluri33 — APPROVED** [review]
approved

File: projects/exui-common-lib/src/lib/services/feature-toggle/feature-toggle.guard.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+    /**
+     * Usage: Add the FeatureToggleGuard as the Guard for a route, and provide the following
+     * in the data array for the route:
+     * - needsFeaturesEnabled: An array of feature keys that need to be enabled for this route
+     * - featureDisabledRedirect: the URL to redirect to when the this route is not accessible due to disabled features
+     * @param route Automatically provided by Angular
+     */
+    public canActivate(route: ActivatedRouteSnapshot): Observable<boolean | UrlTree> {
+        return combineLatest(...(route.data.needsFeaturesEnabled as string[]).map(
```
Comment (andywilkinshmcts) [inline projects/exui-common-lib/src/lib/services/feature-toggle/feature-toggle.guard.ts:23]: Where is route.data.needsFeaturesEnabled initialised? I couldn't see any code in this repo that does it

**Review by andywilkinshmcts — CHANGES_REQUESTED** [review]
Please clarify where route.data.needsFeaturesEnabled is initialised

File: projects/exui-common-lib/src/lib/services/feature-toggle/feature-toggle.guard.ts (hunk 1 above)
Comment (ronaldmansveld) [reply projects/exui-common-lib/src/lib/services/feature-toggle/feature-toggle.guard.ts:23]: It is initialized in the route configuration block, where also the URL, component and guard are set for a route. See the comment above the method.

**Review by andywilkinshmcts — APPROVED** [review]

**Pull request merged**